File: solutions/day1_1.mjs
import * as fs from "fs";

console.log(
    fs
        .readFileSync("/Users/vincent/Documents/AoC23/inputs/input1.txt", { encoding: "utf-8", flag: "r" })
        .trim()
        .split("\n")
        .map((l, i) => l.split("").reduce((acc, c) => acc + (c >= "0" && c <= "9" ? c : ""), ""))
        .map((n) => parseInt(n[0] + n[n.length - 1]))
        .reduce((acc, n) => acc + n, 0)
);

File: solutions/day1_1.py
with open('/Users/vincent/Documents/AoC23/inputs/input1.txt', 'r') as f:
    data = f.read().strip()

def part1(data : str) -> int:
    ans = 0

    for line in data.split():
        first = ""
        last = ""
        for c in line:
            if c.isdigit() and first == "":
                first = c
            if c.isdigit():
                last = c

        num = int(first + last)
        ans += num


    return ans

if __name__ == "__main__":
    print(part1(data))